src/main/java/org/usfirst/frc/team2473/framework/CHS_SparkMax.java

```diff
@@ -0,0 +1,46 @@
+package org.usfirst.frc.team2473.framework;
+
+import com.revrobotics.CANEncoder;
+import com.revrobotics.CANSparkMax;
+import com.revrobotics.CANSparkMaxLowLevel.MotorType;
+
+public class CHS_SparkMax {
+	private CANSparkMax spark;
+	private CANEncoder encoder;
+
+	private double lastEncoderValue = 0;
+
+	public CHS_SparkMax(int port, MotorType motorType) {
+		spark = new CANSparkMax(port, motorType);
+		encoder = spark.getEncoder();
+	}
+
+	public CANSparkMax getSparkMaxObject() {
+		return spark;
+	}
+
+	public void set(double power) {
+		spark.set(power);
+	}
+
+	public void stop() {
+		spark.set(0);
+	}
+
+	public double getEncoderPosition() {
+		if (encoder.getPosition() == 0) {
```

Comment:
When is this ever 0 when lastEncoderValue is non-zero?

Reply:
The Spark MAX controllers occasionally incorrectly report 0 as the encoder position. The encoder count then returns to a normal value. The code that takes the last nonzero value for encoders is to prevent this issue from causing any problems.